[BUG] Building with pre-generated codegen fails

### Reproduction steps

1. Pre-generate code using `scripts/run_in_build_env.sh "scripts/codepregen.py ./zzz_pregenerated/"`
2. Generate Ninja build files with `scripts/build/gn_gen.sh --args="chip_code_pre_generated_directory=\"$(pwd)/zzz_pregenerated\""`
3. Build using `scripts/run_in_build_env.sh "ninja -C ./out chip-repl"`


### Bug prevalence

On every build where chip_code_pre_generated_directory is used

### GitHub hash of the SDK that was being used

9100f280f6a8e7e18c0c388b45c6e81b5f370828

### Platform

python

### Platform Version(s)

_No response_

### Anything else?

The build error printed is:

```
[741/800] ACTION //src/controller/data_model:data_model_codegen_generate(//build/toolchain/linux:linux_x64_gcc)
FAILED: [code=1] gen/src/controller/data_model/app/PluginApplicationCallbacks.h gen/src/controller/data_model/app/callback-stub.cpp gen/src/controller/data_model/app/cluster-callbacks.cpp
python3 ../third_party/pigweed/repo/pw_build/py/pw_build/python_runner.py --gn-root ../ --current-path ../src/controller/data_model --default-toolchain=//build/toolchain/linux:linux_x64_gcc --current-toolchain=//build/toolchain/linux:linux_x64_gcc --capture-output --python-virtualenv-config python/gen/matter_build_venv/venv_metadata.json --python-dep-list-files gen/src/controller/data_model/data_model_codegen_generate_metadata_path_list.txt -- ../scripts/copyfiles.py --source-dir /home/sag/projects/project-chip/connectedhomeip-v1.4.2/zzz_pregenerated/src/controller/data_model/controller-clusters/codegen/cpp-app --target-dir //out/gen/src/controller/data_model /home/sag/projects/project-chip/connectedhomeip-v1.4.2/zzz_pregenerated/src/controller/data_model/controller-clusters/codegen/cpp-app/app/PluginApplicationCallbacks.h /home/sag/projects/project-chip/connectedhomeip-v1.4.2/zzz_pregenerated/src/controller/data_model/controller-clusters/codegen/cpp-app/app/callback-stub.cpp /home/sag/projects/project-chip/connectedhomeip-v1.4.2/zzz_pregenerated/src/controller/data_model/controller-clusters/codegen/cpp-app/app/cluster-callbacks.cpp
Traceback (most recent call last):
  File "/home/sag/projects/project-chip/connectedhomeip-v1.4.2/out/../scripts/copyfiles.py", line 116, in <module>
    main(auto_envvar_prefix='CHIP')
    ~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/sag/projects/project-chip/connectedhomeip-v1.4.2/out/python-venv/lib/python3.13/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
           ~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "/home/sag/projects/project-chip/connectedhomeip-v1.4.2/out/python-venv/lib/python3.13/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
  File "/home/sag/projects/project-chip/connectedhomeip-v1.4.2/out/python-venv/lib/python3.13/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/sag/projects/project-chip/connectedhomeip-v1.4.2/out/python-venv/lib/python3.13/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
  File "/home/sag/projects/project-chip/connectedhomeip-v1.4.2/out/../scripts/copyfiles.py", line 76, in main
    os.makedirs(target_dir)
    ~~~~~~~~~~~^^^^^^^^^^^^
  File "<frozen os>", line 218, in makedirs
  File "<frozen os>", line 218, in makedirs
  File "<frozen os>", line 218, in makedirs
  [Previous line repeated 1 more time]
  File "<frozen os>", line 228, in makedirs
PermissionError: [Errno 13] Permission denied: '//out'
[743/800] c++ obj/third_party/perfetto/repo/sdk/sdk.perfetto.cc.o
ninja: build stopped: subcommand failed.
```

--- a/build/chip/chip_codegen.gni
+++ b/build/chip/chip_codegen.gni
@@ -334,7 +334,7 @@ template("chip_codegen") {
         "--source-dir",
         _generation_dir,
         "--target-dir",
-        target_gen_dir,
+        rebase_path(target_gen_dir, root_build_dir),
       ]
 
       inputs = []

